src/crimson/os/seastore/async_cleaner.cc

```diff
@@ -1218,6 +1192,63 @@ AsyncCleaner::mount_ret AsyncCleaner::mount()
   });
 }
 
+AsyncCleaner::mount_ret AsyncCleaner::_mount_cbjournal()
+{
+  Journal* journal = ecb->get_journal();
+  ceph_assert(journal);
+  auto j = static_cast<journal::CircularBoundedJournal*>(journal);
+  return j->open_device_read_header(
+  ).safe_then([this, j](auto j_seq) {
+    journal_tail_committed = j_seq;
+    update_latest_journal_head_committed(j_seq);
+    update_journal_tail_target(
+      j->get_journal_tail(),
+      j->get_alloc_info_replay_from());
+    return mount_ertr::now();
+  }).handle_error(
+    crimson::ct_error::input_output_error::pass_further{},
+    crimson::ct_error::assert_all{"unexpected error"}
+  );
+}
+
+AsyncCleaner::mount_ret AsyncCleaner::mount()
+{
+  LOG_PREFIX(AsyncCleaner::mount);
+  const auto& sms = sm_group->get_segment_managers();
+  INFO("{} segment managers", sms.size());
+  init_complete = false;
+  stats = {};
+  journal_tail_target = JOURNAL_SEQ_NULL;
+  journal_tail_committed = JOURNAL_SEQ_NULL;
```

Comment:
In my opinion, in case of CBJournal, we need jouranl_tail at the header to know where CBJournal begins because there is no a fixed start position (please see the following example).

CBJournal layout
| header |---------------------------------------|
<<<<<<<<<<<<<<^ this position can be journal tail

Reply:
If you meant only journal_tail_committed in async_cleaner, not journal_tail in CBJournal, I agree

Reply:
> In my opinion, in case of CBJournal, we need jouranl_tail at the header to know where CBJournal begins because there is no a fixed start position

This doesn't persuade me that it is different, `segment_header_t::journal_tail` is also not a fixed position.

`journal_tail_committed` is not the real journal tail in SeaStore, it is a starting point so we can find out the real dirty/alloc tails.

Reply:
I meant that we don't know where the first valid record is started in CBJournal without journal_tail in CBJournal because CBJournal is similar to circular queue. For example,

1. | header | start record | .... |

A record can begin at the position right after the header

2. | header | invalid records... | start record |

But, in case 2, there are invalid records (trimmed), we don't know the start record position, which is valid. Am I wrong?

Reply:
No, you are not wrong.

I just meant `cbj_header_t::journal_tail` is identical to `segment_header_t::journal_tail`.

Seems you might misunderstood what I meant by "getting rid of `journal_tail_committed`" in the beginning. I'm not saying that we don't need it in the header. I meant we might be able to simplify the Cleaner without this in-memory `journal_tail_committed` member.

Reply:
Understood.